spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/test/java/org/springframework/cloud/function/adapter/gcloud/integration/CloudFunctionMain.java

```diff
@@ -32,5 +32,25 @@ public static void main(String[] args) {
 	public Function<String, String> uppercase() {
 		return input -> input.toUpperCase();
 	}
+
+	@Bean
+	public Function<Foo, Bar> foobar() {
+		return input -> new Bar(input.value);
+	}
+}
+
+class Foo {
```

Comment:
Could you make these classes `private static` nested classes? I.e. only have 1 top-level class definition per file.